src/main/java/campus/infrastructure/Parser.java

```diff
@@ -0,0 +1,219 @@
+import campus.tasks.ToDos;
+
+
+/**
+ * Parser handles all the logic for when a command is entered into the ChatBot, it serves as the bridge between the
+ * storage object and taskList object and is the main handler for data travelling between these two fields
+ */
+public class Parser {
+    private final Ui ui;
+    private TaskList taskList;
+    private final Storage storage;
+
+    /**
+     * Initialiser for Paser class
+     * @param ui The Ui Class
+     * @param taskList The TaskList Class
+     * @param storage The Storage Class
+     */
+    public Parser(Ui ui, TaskList taskList, Storage storage) {
+        this.ui = ui;
+        this.storage = storage;
+
+        this.taskList = taskList;
+        this.taskList.updateListFromFile(storage.getListOfStrings());
+    }
+
+    /**
+     * Parses the user's message, works as the main logic tree to handle the user's commands
+     */
+    public String parseMessage(String userInput) {
+        try {
+            String message = this.processCommand(userInput);
+            assert (message != null);
+            return message;
+        } catch (CampusException e) {
+            return this.ui.displayErrorMessage(e);
+        }
+    }
+
+    /**
+     * Process the command fed by the user and decides which path of execution to go next, also handles null cases
+     * and unknown command cases
+     * @param userInput the user command input
+     * @return boolean true for continue feeding, false only in the case of "bye" that is to exit the ChatBot
+     * @throws CampusException Exception in the case that the command is not understood
+     */
+    public String processCommand(String userInput) throws CampusException {
+        String msg = null;
+        String[] arr = userInput.trim().split(" ", 2);
+        String firstWord = arr[0].trim();
+        String remaining = arr.length > 1 ? arr[1].trim() : "";
+
+        switch (firstWord) {
+        case "list":
+            // fallthrough
+        case "mark":
+            // fallthrough
+        case "unmark":
+            // fallthrough
+        case "delete":
+            msg = handleUpdateCommands(firstWord, userInput);
+            break;
+        case "todo":
+            // fallthrough
+        case "deadline":
+            // fallthrough
+        case "event":
+            msg = handleTaskCommand(firstWord, remaining);
+            break;
+        case "bye":
+            return ui.printExiting();
+        case "":
+            break;
+        case "find":
+            msg = handleFindCommand(remaining);
+            break;
+        default:
+            throw new CampusException("Sorry, I don't understand that command,"
+                    + " please check for potential spelling errors");
+        }
+        this.storage.updateFileFromList(this.taskList);
+        assert (msg != null);
+        return msg;
+    }
+
+    private String handleTaskCommand(String command, String details) throws CampusException {
```

Comment:
i like how you named your handle methods

Reply:
thank you!